Lesson19/serkan/script.js

```diff
@@ -0,0 +1,108 @@
+class Person{
+  constructor(name,age,gender){
+      this._name=name;
+      this._age=age;
+      this._gender=gender;
+  }
+  greet(){
+      console.log('Hello my name is'+this._name+'I\'m '+this._age+' years old'+'and my gender is '+this._gender);
+  }
+}
+class Student extends Person{
+  constructor(name,age,gender,major){
+      super(name,age,gender);
+      this._major=major;
+  }
+  greet(){
+      console.log('Hello my name is'+this._name+'I\'m '+this._age+' years old'+'and my gender is '+this._gender+'And I\'m studying '+this._major);
+  }
+}
+
+const person1=new Person('Duygu',32,'Female');
+const person2=new Student('DuyguJr.',23,'Female','Computer Science');
+
+console.log(person1);
+console.log(person2);
+person2.greet();
+
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape to the console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class and adds sideLength and area properties. Override the describe() method in the Square class to log a description of the square to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
+
+class Shape{
+  constructor(name,color) {
+    this._name=name;
+    this._color=color;
+  }
+
+  describe(){
+    console.log('This is a '+this._color+' '+this._name);
+}
+}
+
+class Square extends Shape{
+  constructor(color,sideLength){
+    super('square',color);
+    this._sideLength=sideLength;
+    this._area=sideLength * sideLength;
+  }
+
+  describe(){
+    console.log(`This is a ${this._color} square with a side length  of ${this._sideLength} and an area of ${this._area}`);
+  }
+
+  describeShape () {
+    return super.describe();
+  }
+}
+
+
+
+const square1=new Shape('Square','blue');
+square1.describe();
+console.log (square1);
+
+const square2=new Square('red',5);
+square2.describe();
+square2.describeShape();
+
+
+
+
+// 3. Create a Vehicle class with make, model, and year properties, and a start() method
+// that logs a message to the console indicating that the vehicle has started. Then create a Car class that extends the Vehicle class and adds a numWheels property. Override the start() method in the Car class to log a different message to the console indicating that the car has started.
+class Vehicle {
+  constructor(make,model,year){
+    this._make=make;
+    this._model=model;
+    this._year=year;
+  }
+
+  start(){
+    console.log(`This car has features like ${this._make}  ${this._model}  ${this._year}`);
+  }
+}
+class Car extends Vehicle {
+  constructor () {
+    super (make,model,year,numWheels);
+    this._numWheels=numWheels;
+  }
+  start(){
+    console.log(`This car has features like ${this._make}  ${this._model}  ${this._year} ${this._numWheels}`);
+  }
+
+}
+
+const vehicle1=new Vehicle('Ford', 'Focus', 2020);
+const vehicle2=new Vehicle('Ford', 'Focus', 2020, 4);
```

Comment:
This should be class Car, not vehicle.
```suggestion
const vehicle2=new Car('Ford', 'Focus', 2020, 4);
```